libraries/Microsoft.Bot.Builder/SimplePropertyAccessor.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder
+{
+    public class SimplePropertyAccessor<T> : IPropertyAccessor<T>
+    {
+        private IPropertyContainer propertyContainer;
+        private T defaultValue;
+
+        public SimplePropertyAccessor(IPropertyContainer propertyContainer, string name, T defaultValue = default(T))
+        {
+            this.propertyContainer = propertyContainer;
+            this.Name = name;
+            this.defaultValue = defaultValue;
+        }
+
+        public string Name { get; private set; }
+
+        public Task DeleteAsync(ITurnContext turnContext)
+        {
+            return this.propertyContainer.DeletePropertyAsync(turnContext, this.Name);
+        }
+
+        public async Task<T> GetAsync(ITurnContext turnContext)
+        {
+            T result = await this.propertyContainer.GetPropertyAsync<T>(turnContext, this.Name);
+            if (result == null)
+            {
```

Comment:
You don't want to be in this business. See comments on constructor about taking `Func<T>` instead and remove all of this responsibility from the framework.